docs/resources/cluster_queue.md

```diff
@@ -0,0 +1,74 @@
+# Resource: cluster_queue
+
+This resource allows you to create and manage cluster queues.
+
+Buildkite Documentation: https://buildkite.com/docs/clusters/manage-clusters#set-up-clusters-create-a-queue
+
+## Example Usage
+
+```hcl
+provider "buildkite" {}
+
+resource "buildkite_cluster_queue" "queue1" {
+  cluster_id = "Q2x1c3Rlci0tLTMzMDc0ZDhiLTM4MjctNDRkNC05YTQ3LTkwN2E2NWZjODViNg=="
+  key = "prod-deploy"
+  description = "Prod deployment cluster queue"
+}
+```
+
+## Argument Reference
+
+* `cluster_id` - (Required) The ID of the cluster that this cluster queue belongs to.
+* `key` - (Required) The key of the cluster queue.
+* `description` - (Optional) The description of the cluster queue.
+
+## Attribute Reference
+
+* `id` - The Graphql ID of the created cluster queue.
+* `uuid` - The UUID of the created cluster queue.
+* `cluster_uuid` - The UUID of the cluster that this cluster queue belongs to.
+
+## Import
+
+Cluster queues can be imported using its `GraphQL ID`, along with its respective cluster `UUID`, separated by a comma. e.g.
+
+```
+$ terraform import buildkite_cluster_queue.test Q2x1c3RlclF1ZXVlLS0tYjJiOGRhNTEtOWY5My00Y2MyLTkyMjktMGRiNzg3ZDQzOTAz,35498aaf-ad05-4fa5-9a07-91bf6cacd2bd 
```

Comment:
The ideal use-case here is only 1 ID. And ideally one of the computed fields on the resource. Thats because the other fields required to read the resource should be defined in terraform before you import it anyway. Eg:

```hcl
resource "buildkite_cluster_queue" "test" {
  cluster_id = "<id>"
  key = "main"
  description = "main pipelines"
}
```

Then you simply run `terraform import buildkite_cluster_queue.test Q2x1c3RlclF1ZXVlLS0tYjJiOGRhNTEtOWY5My00Y2MyLTkyMjktMGRiNzg3ZDQzOTAz` and the `ImportState` function uses the `cluster_id` defined in the `resource.ImportStateRequest`.

Reply:
Discussed offline: we'll look to improve this when refactoring to the node query when its implemented server side